bootstrap/src/basis/file.mli

```diff
@@ -0,0 +1,156 @@
+open Rudiments
+
+module Error : sig
+  type t
+
+  val to_string: t -> String.t
+end
+
+module Flag: sig
+  type t =
+  | R_O   (** open existing file for read, failing if it does not exist *)
+  | W     (** create new or truncate existing file and open for write *)
+  | W_A   (** create new or append to existing file and open for write *)
+  | W_AO  (** append to existing file and open for write, failing if it does not
+              exist *)
+  | W_C   (** create new file and open for write, failing if file exists *)
+  | W_O   (** truncate existing file and open for write, failing if file does
+              not exist *)
+  | RW    (** create new or truncate existing file and open for read and write *)
+  | RW_A  (** create new or append to existing file and open for read and write
+              *)
+  | RW_AO (** append to existing file and open for read and write, failing if
+              file does not exist *)
+  | RW_C  (** create new file and open for read and write, failing if file
+              exists *)
+  | RW_O  (** truncate existing file and open for read and write, failing if
+              file does not exist *)
+end
+
+module Buffer: sig
+  type t = Byte.t Array.Cursor.t * Byte.t Array.Cursor.t
+
+  val of_bytes: Bytes.t -> t
+end
+
+type t
+(** mutable Unix file descriptor *)
+
+(* val of_path: Flag.t -> usize -> Bytes.t !-> (?t, Error.t) result *)
+val of_path: ?flag:Flag.t -> ?mode:usize -> Bytes.t -> (t, Error.t) result
+(** [of_path mode path] opens file in [mode] at [path] and returns the resulting
+    file or an error if file could not be opened. *)
+
+(* val of_path_hlt: Flag.t -> usize -> Bytes.t !-> ?t *)
+val of_path_hlt: ?flag:Flag.t -> ?mode:usize -> Bytes.t -> t
+(** [of_path mode path] opens file in [mode] at [path] and returns the resulting
+    file or halts if file could not be opened. *)
+
+(* val of_stdin: unit !-> Bytes.t !-> ?t *)
```

Comment:
```suggestion
(* val of_stdin: unit !-> Bytes.t !-> t *)
```
The `file` isn't necessarily isolated; it could be a copy of a value in the environment.  Same for `stdout` and `stderr`.  In fact, these should probably be much simpler (values rather than functions).
```suggestion
(* val stdin: t *)
```